src/common/verify/isValidAccessGrant.test.ts

```diff
@@ -184,6 +248,32 @@ describe("isValidAccessGrant", () => {
     expect(mockedFetch).toHaveBeenCalledWith("https://example.com/someVc");
   });
 
+  it("retrieves the vc if a plain json from url was passed", async () => {
+    jest.mocked(isVerifiableCredential).mockReturnValueOnce(true);
+    const mockedFetch = jest
+      .fn<typeof fetch>()
+      // First, the VC is fetched
+      .mockResolvedValueOnce(
+        new Response(JSON.stringify(MOCK_ACCESS_GRANT), { status: 200 }),
+      )
+      // Then, the verification endpoint is called
+      .mockResolvedValueOnce(
+        new Response(JSON.stringify(MOCK_VERIFY_RESPONSE), {
+          status: 200,
+        }),
+      );
+
+    await isValidAccessGrant(
+      JSON.parse(JSON.stringify("https://example.com/someVc")),
+      {
+        fetch: mockedFetch,
+        verificationEndpoint: MOCK_ACCESS_ENDPOINT,
+      },
+    );
+
+    expect(mockedFetch).toHaveBeenCalledWith("https://example.com/someVc");
+  });
```

Comment:
```suggestion
```
In this test, it's not an Access Grant that is passed, but a URL, so we are already fetching and parsing the VC as we see fit. There is nothing to add to the original test.